cmake/FindHiopHipLibraries.cmake

```diff
@@ -0,0 +1,52 @@
+#[[
+
+Exports target `hiop_hip` which finds all hip libraries needed by hiop.
+
+#]]
+
+add_library(hiop_hip INTERFACE)
+
+find_library(HIPBLAS_LIBRARY
+  NAMES
+  hipblas
+  PATHS
+  ENV LD_LIBRARY_PATH ENV DYLD_LIBRARY_PATH
+  PATH_SUFFIXES
+  lib64 lib)
+
+if(HIPBLAS_LIBRARY)
+  get_filename_component(HIPBLAS_LIBRARY_DIR ${HIPBLAS_LIBRARY} DIRECTORY)
+endif()
+
+find_path(HIPBLAS_INCLUDE_DIR
+  NAMES
+  hipblas.h
+  PATHS
+  ${HIPBLAS_LIBRARY_DIR}/..
+  PATH_SUFFIXES
+  include)
+
+if(HIPBLAS_LIBRARY)
+  add_library(Hipblas INTERFACE)
+  target_link_libraries(Hipblas INTERFACE ${HIPBLAS_LIBRARY})
+  target_include_directories(Hipblas INTERFACE ${HIPBLAS_INCLUDE_DIR})
+  message(STATUS "Found Hipblas include: ${HIPBLAS_INCLUDE_DIR}")
+  message(STATUS "Found Hipblas library: ${HIPBLAS_LIBRARY}")
```

Comment:
Since this is a dependency that is not a part of an export set, we should (similar to COINHSL [here](https://github.com/LLNL/hiop/commit/8916a52daae1cb73d011089a66516682fb2f0e4d#diff-ea5d33bbfc6af651063ada4999653ea00a02dc47c347e5304356b6af6ed00281R44)) add:
```CMake
install(TARGETS Hipblas EXPORT hiop-targets)
```
Another option is to add logic in `cmake/HiOpConfig.cmake.in`, however for Hipblas specifically what I suggested seems best.